zerver/webhooks/patreon/view.py

```diff
@@ -0,0 +1,116 @@
+from typing import Any, Callable, Dict, Optional
+
+from django.http import HttpRequest, HttpResponse
+
+from zerver.decorator import webhook_view
+from zerver.lib.exceptions import UnsupportedWebhookEventType
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.lib.webhooks.common import (
+    check_send_webhook_message,
+    validate_extract_webhook_http_header,
+)
+from zerver.models import UserProfile
+
+
+# convert fixture names of type members_create -> member:create
+# this is done to match the http header X_PATREON_EVENT we get from the incoming webhook
+# check patreon webhook docs for more info on why we implemented this
+def get_http_headers_from_filename(http_header_key: str) -> Callable[[str], Dict[str, str]]:
+    def fixture_to_headers(filename: str) -> Dict[str, str]:
+        event_type = ":".join(filename.split("_"))
+        return {http_header_key: event_type}
+
+    return fixture_to_headers
+
+
+fixture_to_headers = get_http_headers_from_filename("HTTP_X_PATREON_EVENT")
+
+# called when new patron joined
+def members_create(payload: Dict[str, Any]) -> str:
+    included = payload["included"][0]
+    patron_count = included["attributes"]["patron_count"]
+    body = f"New patron joined! :tada:\nPatrons in total: {patron_count}"
```

Comment:
Can this show the identity and patron tier?

Reply:
I could find the name of the new patron from the fixture. However the tier info is not there in the response. We can show the amount paid by the patron. There was not much documentation provided in the official docs for this. I found this blog trying to solve a similar problem [here](https://www.brianchuchua.dev/patreon-webhooks/). We could show the tier either by:

1. Comparing the amount paid and assign a tier or
2. Comparing the tier ID

Either way I think we should wait before implementing this. The patreon docs are pretty confusing. I think we should test out first this webhook with live data.